src/test/java/com/pgs/booking/service/FlightServiceTest.java

```diff
@@ -0,0 +1,162 @@
+package com.pgs.booking.service;
+
+import com.pgs.booking.exception.ResourceNotFoundException;
+import com.pgs.booking.mappers.CreateUpdateFlightDtoMapper;
+import com.pgs.booking.mappers.FlightDtoMapper;
+import com.pgs.booking.model.Flight;
+import com.pgs.booking.model.dto.CreateUpdateFlightDto;
+import com.pgs.booking.model.dto.FlightDto;
+import com.pgs.booking.repository.FlightRepository;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.util.List;
+import java.util.Optional;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class FlightServiceTest {
+
+    @Mock
+    private FlightRepository flightRepository;
+
+    private final FlightDtoMapper flightDtoMapper = new FlightDtoMapper();
+    private final CreateUpdateFlightDtoMapper createUpdateFlightDtoMapper = new CreateUpdateFlightDtoMapper();
+
+    @InjectMocks
```

Comment:
not needed, as we're injecting mock manually in constructor

Reply:
You have 2 options here (both valid):

- remove `@InjectMocks` as the service is initialised in setUp()
- remove `setUp()` and add `@Spy` on `flightDtoMapper` and `createUpdateFlightDtoMapper` to make `@InjectMocs` work with non-mock fields